### ISSUE TITLE
3.0: build fails with gcc 13

### ISSUE THREAD

**Issue by kloczek** [oneapi-src/oneDNN#1548]
Looks like 3.0 id not ready for gcc 13.0.1 which s availabe in fedora rawhide.
```console
[ 47%] Building CXX object src/gpu/jit/CMakeFiles/dnnl_gpu_jit.dir/gemm/kernel_evaluator.cpp.o
cd /home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/x86_64-redhat-linux-gnu/src/gpu/jit && /usr/bin/g++ -DCL_TARGET_OPENCL_VERSION=120 -DDNNL_DLL -DDNNL_DLL_EXPORTS -DDNNL_ENABLE_CPU_ISA_HINTS -DDNNL_ENABLE_ITT_TASKS -DDNNL_ENABLE_MAX_CPU_ISA -DDNNL_X64=1 -DNGEN_CPP11 -DNGEN_NEO_INTERFACE -DNGEN_NO_OP_NAMES -DNGEN_SAFE -DNGEN_WINDOWS_COMPAT -D__STDC_CONSTANT_MACROS -D__STDC_LIMIT_MACROS -I/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/include -I/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/x86_64-redhat-linux-gnu/include -I/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src -O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none -fopenmp -fvisibility-inlines-hidden  -Wall -Wno-unknown-pragmas -fvisibility=internal -Wno-ignored-attributes -msse4.1  -fPIC -Wformat -Wformat-security -fstack-protector-strong  -Wmissing-field-initializers -Wmissing-field-initializers  -Wno-strict-overflow -Wno-maybe-uninitialized  -DNDEBUG -std=c++11 -MD -MT src/gpu/jit/CMakeFiles/dnnl_gpu_jit.dir/gemm/kernel_evaluator.cpp.o -MF CMakeFiles/dnnl_gpu_jit.dir/gemm/kernel_evaluator.cpp.o.d -o CMakeFiles/dnnl_gpu_jit.dir/gemm/kernel_evaluator.cpp.o -c /home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/kernel_evaluator.cpp
In file included from /home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/kernel_catalog.hpp:24,
                 from /home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/kernel_evaluator.hpp:20,
                 from /home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/kernel_evaluator.cpp:17:
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:30:15: error: found ‘:’ in nested-name-specifier, expected ‘::’
   30 | enum LoopType : uint8_t {
      |               ^
      |               ::
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:30:6: error: ‘LoopType’ has not been declared
   30 | enum LoopType : uint8_t {
      |      ^~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:30:25: error: expected unqualified-id before ‘{’ token
   30 | enum LoopType : uint8_t {
      |                         ^
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:49:13: error: found ‘:’ in nested-name-specifier, expected ‘::’
   49 | enum WGType : uint8_t {
      |             ^
      |             ::
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:49:6: error: ‘WGType’ has not been declared
   49 | enum WGType : uint8_t {
      |      ^~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:49:23: error: expected unqualified-id before ‘{’ token
   49 | enum WGType : uint8_t {
      |                       ^
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:58:5: error: ‘LoopType’ does not name a type
   58 |     LoopType fusedLoop; // Loop dimension in which EUs are fused (if any).
      |     ^~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:60:5: error: ‘LoopType’ does not name a type
   60 |     LoopType loopOrder
      |     ^~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:67:5: error: ‘WGType’ does not name a type
   67 |     WGType wgUpdate; // Work group type showing how/if work group sizes can be updated.
      |     ^~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:103:16: error: ‘LoopType’ has not been declared
  103 |     int wgTile(LoopType l) const { return unroll[l] * wg[l]; }
      |                ^~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp: In member function ‘bool dnnl::impl::gpu::jit::CommonDriverInfo::fusedEUs() const’:
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:78:37: error: ‘fusedLoop’ was not declared in this scope
   78 |     bool fusedEUs() const { return (fusedLoop != LoopNone); }
      |                                     ^~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:78:50: error: ‘LoopNone’ was not declared in this scope
   78 |     bool fusedEUs() const { return (fusedLoop != LoopNone); }
      |                                                  ^~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp: In member function ‘bool dnnl::impl::gpu::jit::CommonDriverInfo::isMNK() const’:
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:80:18: error: ‘loopOrder’ was not declared in this scope
   80 |         auto l = loopOrder[0] & ~LoopPersistent;
      |                  ^~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:80:34: error: ‘LoopPersistent’ was not declared in this scope; did you mean ‘isPersistent’?
   80 |         auto l = loopOrder[0] & ~LoopPersistent;
      |                                  ^~~~~~~~~~~~~~
      |                                  isPersistent
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:81:21: error: ‘LoopM’ was not declared in this scope
   81 |         return l == LoopM || l == LoopMNHilbertMNK
      |                     ^~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:81:35: error: ‘LoopMNHilbertMNK’ was not declared in this scope
   81 |         return l == LoopM || l == LoopMNHilbertMNK
      |                                   ^~~~~~~~~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:82:25: error: ‘LoopMNBoustrophedonMNK’ was not declared in this scope
   82 |                 || l == LoopMNBoustrophedonMNK;
      |                         ^~~~~~~~~~~~~~~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp: In member function ‘bool dnnl::impl::gpu::jit::CommonDriverInfo::isNMK() const’:
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:85:18: error: ‘loopOrder’ was not declared in this scope
   85 |         auto l = loopOrder[0] & ~LoopPersistent;
      |                  ^~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:85:34: error: ‘LoopPersistent’ was not declared in this scope; did you mean ‘isPersistent’?
   85 |         auto l = loopOrder[0] & ~LoopPersistent;
      |                                  ^~~~~~~~~~~~~~
      |                                  isPersistent
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:86:21: error: ‘LoopN’ was not declared in this scope
   86 |         return l == LoopN || l == LoopMNHilbertNMK
      |                     ^~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:86:35: error: ‘LoopMNHilbertNMK’ was not declared in this scope
   86 |         return l == LoopN || l == LoopMNHilbertNMK
      |                                   ^~~~~~~~~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:87:25: error: ‘LoopMNBoustrophedonNMK’ was not declared in this scope
   87 |                 || l == LoopMNBoustrophedonNMK;
      |                         ^~~~~~~~~~~~~~~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp: In member function ‘bool dnnl::impl::gpu::jit::CommonDriverInfo::isHilbert() const’:
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:90:18: error: ‘loopOrder’ was not declared in this scope
   90 |         auto l = loopOrder[0] & ~LoopPersistent;
      |                  ^~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:90:34: error: ‘LoopPersistent’ was not declared in this scope; did you mean ‘isPersistent’?
   90 |         auto l = loopOrder[0] & ~LoopPersistent;
      |                                  ^~~~~~~~~~~~~~
      |                                  isPersistent
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:91:21: error: ‘LoopMNHilbertMNK’ was not declared in this scope
   91 |         return l == LoopMNHilbertMNK || l == LoopMNHilbertNMK;
      |                     ^~~~~~~~~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:91:46: error: ‘LoopMNHilbertNMK’ was not declared in this scope
   91 |         return l == LoopMNHilbertMNK || l == LoopMNHilbertNMK;
      |                                              ^~~~~~~~~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp: In member function ‘bool dnnl::impl::gpu::jit::CommonDriverInfo::isBoustrophedon() const’:
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:94:18: error: ‘loopOrder’ was not declared in this scope
   94 |         auto l = loopOrder[0] & ~LoopPersistent;
      |                  ^~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:94:34: error: ‘LoopPersistent’ was not declared in this scope; did you mean ‘isPersistent’?
   94 |         auto l = loopOrder[0] & ~LoopPersistent;
      |                                  ^~~~~~~~~~~~~~
      |                                  isPersistent
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:95:21: error: ‘LoopMNBoustrophedonMNK’ was not declared in this scope
   95 |         return l == LoopMNBoustrophedonMNK || l == LoopMNBoustrophedonNMK;
      |                     ^~~~~~~~~~~~~~~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:95:52: error: ‘LoopMNBoustrophedonNMK’ was not declared in this scope
   95 |         return l == LoopMNBoustrophedonMNK || l == LoopMNBoustrophedonNMK;
      |                                                    ^~~~~~~~~~~~~~~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp: In member function ‘bool dnnl::impl::gpu::jit::CommonDriverInfo::isPersistent() const’:
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:99:17: error: ‘loopOrder’ was not declared in this scope
   99 |         return (loopOrder[0] != LoopNone) && (loopOrder[0] & LoopPersistent);
      |                 ^~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:99:33: error: ‘LoopNone’ was not declared in this scope
   99 |         return (loopOrder[0] != LoopNone) && (loopOrder[0] & LoopPersistent);
      |                                 ^~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:99:62: error: ‘LoopPersistent’ was not declared in this scope; did you mean ‘isPersistent’?
   99 |         return (loopOrder[0] != LoopNone) && (loopOrder[0] & LoopPersistent);
      |                                                              ^~~~~~~~~~~~~~
      |                                                              isPersistent
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp: In member function ‘bool dnnl::impl::gpu::jit::CommonDriverInfo::fixedWG() const’:
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:101:35: error: ‘wgUpdate’ was not declared in this scope
  101 |     bool fixedWG() const { return wgUpdate == WGFixed; }
      |                                   ^~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_common.hpp:101:47: error: ‘WGFixed’ was not declared in this scope
  101 |     bool fixedWG() const { return wgUpdate == WGFixed; }
      |                                               ^~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/kernel_catalog.hpp: At global scope:
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/kernel_catalog.hpp:199:5: error: ‘uint64_t’ does not name a type
  199 |     uint64_t revision DEFAULT(0);
      |     ^~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/kernel_catalog.hpp:25:1: note: ‘uint64_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
   24 | #include "gen_gemm_kernel_common.hpp"
  +++ |+#include <cstdint>
   25 |
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/kernel_catalog.hpp:208:5: error: ‘uint64_t’ does not name a type
  208 |     uint64_t revision;
      |     ^~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/kernel_catalog.hpp:208:5: note: ‘uint64_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/kernel_catalog.hpp: In member function ‘dnnl::impl::gpu::jit::kcatalog::FlatCatalog<n>::operator dnnl::impl::gpu::jit::kcatalog::Catalog() const’:
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/kernel_catalog.hpp:213:37: error: ‘revision’ was not declared in this scope
  213 |         Catalog catalog = {version, revision, entryCount, &entries[0]};
      |                                     ^~~~~~~~
In file included from /home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/kernel_evaluator.hpp:22:
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp: At global scope:
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp:515:5: error: ‘LoopType’ does not name a type
  515 |     LoopType var; // Variable to base mask off of
      |     ^~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp: In member function ‘bool dnnl::impl::gpu::jit::MaskAssignment::compatible(const dnnl::impl::gpu::jit::MaskAssignment&) const’:
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp:520:38: error: ‘var’ was not declared in this scope
  520 |         return mask == other.mask && var == other.var && offset == other.offset;
      |                                      ^~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp:520:51: error: ‘const struct dnnl::impl::gpu::jit::MaskAssignment’ has no member named ‘var’
  520 |         return mask == other.mask && var == other.var && offset == other.offset;
      |                                                   ^~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp: At global scope:
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp:875:5: error: ‘LoopType’ does not name a type
  875 |     LoopType loopOrder[3] = {LoopM, LoopN,
      |     ^~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp:877:5: error: ‘LoopType’ does not name a type
  877 |     LoopType fusedLoop = LoopM; // Direction of fusing if threads fused.
      |     ^~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp:886:5: error: ‘WGType’ does not name a type; did you mean ‘Type’?
  886 |     WGType forceWGUpdate = WGDynamic; // Force work group update type.
      |     ^~~~~~
      |     Type
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp:1046:5: error: ‘WGType’ does not name a type; did you mean ‘Type’?
 1046 |     WGType getWGType(const GEMMProblem &problem) const {
      |     ^~~~~~
      |     Type
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp: In member function ‘int dnnl::impl::gpu::jit::GEMMStrategy::slmABufBlockSize(const dnnl::impl::gpu::jit::GEMMProblem&) const’:
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp:1012:34: error: ‘LoopM’ was not declared in this scope
 1012 |                         * unroll[LoopM] * maxKSLM(problem, true);
      |                                  ^~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp: In member function ‘int dnnl::impl::gpu::jit::GEMMStrategy::slmBBufBlockSize(const dnnl::impl::gpu::jit::GEMMProblem&) const’:
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp:1017:34: error: ‘LoopN’ was not declared in this scope
 1017 |                         * unroll[LoopN] * maxKSLM(problem, false);
      |                                  ^~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp: In member function ‘int dnnl::impl::gpu::jit::GEMMStrategy::slmABufSize(const dnnl::impl::gpu::jit::GEMMProblem&) const’:
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp:1020:47: error: ‘LoopM’ was not declared in this scope
 1020 |         return slmABufBlockSize(problem) * wg[LoopM] * wg[LoopK] * slmBuffers;
      |                                               ^~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp:1020:59: error: ‘LoopK’ was not declared in this scope
 1020 |         return slmABufBlockSize(problem) * wg[LoopM] * wg[LoopK] * slmBuffers;
      |                                                           ^~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp: In member function ‘int dnnl::impl::gpu::jit::GEMMStrategy::slmBBufSize(const dnnl::impl::gpu::jit::GEMMProblem&) const’:
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp:1023:47: error: ‘LoopN’ was not declared in this scope
 1023 |         return slmBBufBlockSize(problem) * wg[LoopN] * wg[LoopK] * slmBuffers;
      |                                               ^~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp:1023:59: error: ‘LoopK’ was not declared in this scope
 1023 |         return slmBBufBlockSize(problem) * wg[LoopN] * wg[LoopK] * slmBuffers;
      |                                                           ^~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp: In member function ‘int dnnl::impl::gpu::jit::GEMMStrategy::slmSysgemmBlockSize() const’:
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp:1026:26: error: ‘LoopM’ was not declared in this scope
 1026 |         return 1152 * wg[LoopM] + 1536 * wg[LoopN];
      |                          ^~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp:1026:45: error: ‘LoopN’ was not declared in this scope
 1026 |         return 1152 * wg[LoopM] + 1536 * wg[LoopN];
      |                                             ^~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp: In member function ‘bool dnnl::impl::gpu::jit::GEMMStrategy::fusedM() const’:
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp:1043:44: error: ‘fusedLoop’ was not declared in this scope
 1043 |     bool fusedM() const { return fused && (fusedLoop == LoopM); }
      |                                            ^~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp:1043:57: error: ‘LoopM’ was not declared in this scope
 1043 |     bool fusedM() const { return fused && (fusedLoop == LoopM); }
      |                                                         ^~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp: In member function ‘bool dnnl::impl::gpu::jit::GEMMStrategy::fusedN() const’:
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp:1044:44: error: ‘fusedLoop’ was not declared in this scope
 1044 |     bool fusedN() const { return fused && (fusedLoop == LoopN); }
      |                                            ^~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp:1044:57: error: ‘LoopN’ was not declared in this scope
 1044 |     bool fusedN() const { return fused && (fusedLoop == LoopN); }
      |                                                         ^~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp: In member function ‘bool dnnl::impl::gpu::jit::GEMMStrategy::fixedWG(const dnnl::impl::gpu::jit::GEMMProblem&) const’:
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp:1055:17: error: ‘getWGType’ was not declared in this scope
 1055 |         return (getWGType(problem) == WGFixed);
      |                 ^~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp:1055:39: error: ‘WGFixed’ was not declared in this scope
 1055 |         return (getWGType(problem) == WGFixed);
      |                                       ^~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp: At global scope:
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp:1834:58: error: ‘LoopType’ has not been declared
 1834 |     bool assignMasks(std::vector<RegisterBlock> &layout, LoopType rloop,
      |                                                          ^~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp:1835:13: error: ‘LoopType’ has not been declared
 1835 |             LoopType cloop, std::vector<MaskAssignment> &assignments,
      |             ^~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/gen_gemm_kernel_generator.hpp:2248:30: error: ‘LoopType’ has not been declared
 2248 |     bool mnRemainderHandling(LoopType loop, Problem &problem,
      |                              ^~~~~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/kernel_evaluator.cpp: In function ‘double dnnl::impl::gpu::jit::evaluateW(const kcatalog::Entry&, const DerivedEvaluateParams&, EvaluateAuxOutput&)’:
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/kernel_evaluator.cpp:44:60: error: ‘LoopK’ was not declared in this scope
   44 |         aux.k0 = alignUp(divUp(dp.sizes.k, e.driverInfo.wg[LoopK]),
      |                                                            ^~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/kernel_evaluator.cpp:52:66: error: ‘LoopK’ was not declared in this scope
   52 |                 divUp(dp.sizes.k, wgCountK), e.driverInfo.unroll[LoopK]);
      |                                                                  ^~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/kernel_evaluator.cpp: In function ‘double dnnl::impl::gpu::jit::evaluateSCore(const kcatalog::Entry&, const DerivedEvaluateParams&, EvaluateAuxOutput&)’:
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/kernel_evaluator.cpp:84:42: error: ‘LoopK’ was not declared in this scope
   84 |                 divUp(k, e.driverInfo.wg[LoopK]), e.driverInfo.unroll[LoopK]);
      |                                          ^~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/kernel_evaluator.cpp:117:35: error: ‘LoopM’ was not declared in this scope
  117 |     etime *= (e.driverInfo.unroll[LoopM] * e.driverInfo.unroll[LoopN]);
      |                                   ^~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/kernel_evaluator.cpp:117:64: error: ‘LoopN’ was not declared in this scope
  117 |     etime *= (e.driverInfo.unroll[LoopM] * e.driverInfo.unroll[LoopN]);
      |                                                                ^~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/kernel_evaluator.cpp: In function ‘double dnnl::impl::gpu::jit::evaluateS(const kcatalog::Entry&, const DerivedEvaluateParams&, EvaluateAuxOutput&)’:
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/kernel_evaluator.cpp:143:73: error: ‘LoopK’ was not declared in this scope
  143 |                 = alignUp(divUp(dp.sizes.k, wgCountK1 * e.driverInfo.wg[LoopK]),
      |                                                                         ^~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/kernel_evaluator.cpp: In function ‘dnnl::impl::gpu::jit::DerivedEvaluateParams dnnl::impl::gpu::jit::getDerivedParams(const kcatalog::Entry&, const EvaluateParams&)’:
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/kernel_evaluator.cpp:211:40: error: ‘LoopM’ was not declared in this scope
  211 |     auto unrollM = e.driverInfo.unroll[LoopM];
      |                                        ^~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/kernel_evaluator.cpp:212:40: error: ‘LoopN’ was not declared in this scope
  212 |     auto unrollN = e.driverInfo.unroll[LoopN];
      |                                        ^~~~~
/home/tkloczko/rpmbuild/BUILD/oneDNN-3.0/src/gpu/jit/gemm/kernel_evaluator.cpp:237:53: error: ‘LoopK’ was not declared in this scope
  237 |     auto threadsPerWG = wgM * wgN * e.driverInfo.wg[LoopK];
      |                                                     ^~~~~
make[2]: *** [src/gpu/jit/CMakeFiles/dnnl_gpu_jit.dir/build.make:303: src/gpu/jit/CMakeFiles/dnnl_gpu_jit.dir/gemm/kernel_evaluator.cpp.o] Error 1
```

**Comment by vpirogov** [comment]
+@petercad

**Comment by petercad** [comment]
Looks like a missing `#include <cstdint>`. I'll put together a patch shortly.

**Comment by vpirogov** [comment]
Fixed by 940ccf3fe0d2ebb30dbe456831904596a8b0c1b4 and backported to rls-v3.0.
